### ISSUE TITLE
Allow manually setting of MongoDB instead of using Autowired details

### ISSUE THREAD

**Issue by ClaudioConsolmagno** [javers/javers#694]
Currently `JaversMongoAutoConfiguration.java` forces you to use the autowired `MongoClient` (and `MongoProperties`). This is fine if you want to have javers auditing in the same host:DB as your primary mongoDB.

We have a use case where we want to keep the (javers) auditing DB and primary MongoDB separate and for that we want to be able to pass in mongo credentials from the properties file (e.g. with javers.uri) that will cause the autoconfig class to create a `MongoClient` instead of using the autowired instance.

What are the thoughts on this? Would be happy to submit a pull-request.

**Comment by bartoszwalacik** [comment]
Hi. Looks like a good idea, we will be happy to merge your PR

**Comment by shirishpandharikar** [comment]
@ClaudioConsolmagno @bartoszwalacik Any thoughts on this? I was looking into this and it doesn't seem to be a straightforward fix. Essentially we would have to define all configuration properties for MongoDB in `JaversMongoProperties` and inject `MongoDatabase` instead of a `MongoClient`. We also need a configuration property, let's say **`javers-properties-override`** [couldn't find a better name] which if set to **`true`** tells Javers to create its own `MongoClient` and obtain a `MongoDatabase` instance. If set to **`false`**, it will fall back to using the spring-boot-starter-data-mongodb configured instance.

**Comment by bartoszwalacik** [comment]
Indeed, some props have to be added, for example:

```
javers:
    mongodb:
        database: my-db
        host: "localhost"
        ....
```

Maybe it's enough to check if  any of `javers.mongodb.*` props is present instead of using `javers-properties-override`?

**Comment by shirishpandharikar** [comment]
@bartoszwalacik Have a look at the PR and let me know how it looks.

**Comment by bartoszwalacik** [comment]
fixed in 5.6.0
see https://javers.org/documentation/spring-boot-integration/#starter-repository-configuration
